components/Location.tsx

```diff
@@ -41,19 +41,19 @@ function Location({
               setFormData({ ...formData, postcode: e.target.value })
             }
             ref={register({
-              required: "Postcode is required",
+              required: pin || "Field is required",
               pattern: {
-                value: /^[1-9]\d*$/,
+                value: /^[A-Za-z0-9]{1,10}$/,
```

Comment:
Keep regex specific to the country. Find out the format for Canada, US and India. You will find readymade regex for each of these